pcsx2/Frontend/CommonHost.cpp

```diff
@@ -542,4 +549,17 @@ std::string CommonHost::GetRichPresenceString()
 	return rich_presence_string;
 }
 
-#endif
\ No newline at end of file
+#endif
+
+void CommonHost::ReloadPINE()
+{
+	bool successful = true;
+	if (EmuConfig.EnablePINE && (s_pine_server.m_slot != EmuConfig.PINESlot || s_pine_server.m_end))
+	{
+		successful = s_pine_server.Initialize(EmuConfig.PINESlot);
+	}
+	else if ((!EmuConfig.EnablePINE && !s_pine_server.m_end) || !successful)
```

Comment:
Was this intended to be an `if`, so the `successful` assignment above has an effect? Tbh it would be better to call Deinitialize() in the failure branches of Initialize(), that way the object is always in a consistent state upon returning from Initialize().

Reply:
Yes this was broken, and there were more issues around initialization under the hood I noticed when trying to fix it.
Now I make sure to always Deinitialize before initializing if it's running and we need to reload it.
I've also made sure made it so that Deinitialize is called in the failure branches of initialize, good idea.
And I've made sure no funny business can happen even if you call Deinitialize first, or multiple times in a row even (not that we do that in this code, but stability good)